search/includes/classes/queue/class-cron.php

```diff
@@ -114,9 +125,47 @@ public function process_jobs( $job_ids ) {
 		$this->queue->process_jobs( $jobs );
 	}
 
+	/**
+	 * Given a term taxonomy id, queue all posts for reindexing that match it
+	 *
+	 * @param {int} $term_taxonomy_id The term taxonomy id you want to index
+	 */
+	public function queue_posts_for_term_taxonomy_id( $term_taxonomy_id ) {
+		$args = array(
+			'posts_per_page' => self::TERM_UPDATE_BATCH_SIZE,
+			'paged' => 1,
+			'tax_query' => array(
+				array(
+					'field' => 'term_taxonomy_id',
+					'terms' => $term_taxonomy_id,
+				),
+			),
```

Comment:
It's kind of a pain to use WP_Query here instead of a direct query, but we _have_ to because we need to run any filters that have been added, and this helps ensure we return only indexable posts.

Speaking of, are we sure this query will only return posts that are indexable? (either by status, or post type, etc). Would be bad if we accidentally indexed otherwise hidden posts.

Reply:
I'm not sure if EP adds them or not. If not, we'll have to used the indexable/hooks to get the correct filters applied.